### PR TITLE
docs: document PyTorch backend

### LINKED ISSUE #3121: [Feature Request] Documentation for multiple backends
### Summary

Propose a way to document multiple backends.

### Detailed Description

We may use font-awesome icons like [cibuildwheel](https://cibuildwheel.readthedocs.io/en/stable/setup/#github-actions):

image
image

Readers can quickly know which backend is supported.

If commands are different on different backends, Sphinx provides a way:

image

The problem is that TF and PT icons are unavailable in font-awesome, per https://github.com/FortAwesome/Font-Awesome/issues/17377. We need to make them by ourselves...

### Further Information, Files, and Links

_No response_

### PR DESCRIPTION
**Pull request by njzjz**
Fix #3121.

There are TODOs:
(1) PyTorch-backend specific features and arguments;
(2) Python interface installation. Currently, the TensorFlow backend is always installed, and I am considering rewriting the logic;
(3) Unsupported features - write docs when implemented.

### CODE REVIEW

File: doc/install/install-from-source.md [hunk 1]
```diff
@@ -14,45 +14,74 @@ cd deepmd-kit
...
+python -c "import tensorflow as tf;print(tf.reduce_sum(tf.random.normal([1000, 1000])))"
 ```
-If one has multiple python interpreters named something like python3.x, it can be specified by, for example
+
+One can also [build the TensorFlow Python interface from source](https://www.tensorflow.org/install/source) for customized hardware optimization, such as CUDA, ROCM, or OneDNN support.
+
+:::
+
+:::{tab-item} PyTorch {{ pytorch_icon }}
+
```
Comment (anyangml) [inline doc/install/install-from-source.md:58]: it seems some of the special characters are not rendered correctly.

**Review by anyangml — COMMENTED** [review]
LGTM, except the rendering issue.

File: doc/install/install-from-source.md (hunk 1 above)
Comment (njzjz) [reply doc/install/install-from-source.md:58]: I can see it rendered correctly in the preview of the web page. If you mean the page rendered in GitHub, we have deprecated that one.

image

**Review by anyangml — APPROVED** [review]

File: doc/install/install-from-source.md (hunk 1 above)
Comment (anyangml) [reply doc/install/install-from-source.md:58]: Approved.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**